version: '3.7'

services:
  runner1:
    hostname: runner1
    image: gitlab/gitlab-runner:alpine
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3
    volumes:
      - ${GITLAB_RUNNERS_HOME}/runner1:/etc/gitlab-runner 
      - /var/run/docker.sock:/var/run/docker.sock
      - ./scripts:/opt/scripts:ro
    networks:
        - gitlab-network
    entrypoint: 
      - /usr/bin/dumb-init
      - --
    command: 
      - bash
      - -c
      - /opt/scripts/register.sh && /opt/scripts/run.sh
    env_file:
     - .env

  runner2:
    hostname: runner2
    image: gitlab/gitlab-runner:alpine
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3
    volumes:
      - ${GITLAB_RUNNERS_HOME}/runner2:/etc/gitlab-runner 
      - /var/run/docker.sock:/var/run/docker.sock
      - ./scripts:/opt/scripts:ro
    networks:
        - gitlab-network
    entrypoint: 
      - /usr/bin/dumb-init
      - --
    command: 
      - bash
      - -c
      - /opt/scripts/register.sh && /opt/scripts/run.sh
    env_file:
     - .env

networks:
  gitlab-network:
    name: gitlab-network
